### PR TITLE
refactor!: move parts of libsinsp under the chisel directory

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind design

**Any specific area of the project related to this PR?**

/area libsinsp

**What this PR does / why we need it**:

This is a clean-up PR that moves pieces around. `table.h`, `table.cpp`, `viewinfo.h`, `viewinfo.cpp`, `sinsp_capture_interrupt_exception.h` have been moved from `userspace/libsinsp` into the `userspace/chisel` directory because they are used for chisel features only. On top of that, the definitions have been renamed to have the `chisel_` prefix instead of the `sinsp_` one. The end goal is to better separate responsibilities and have a leaner libsinsp codebase.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
refactor!: move parts of libsinsp under the chisel directory
```

### CODE REVIEW

**Review by Molter73 — COMMENTED** [review]
This looks like a pretty good tidying for the chisels, I have a few comments/questions, but overall LGTM!

Also, I don't want to get too deep into naming semantics, but maybe it makes sense to drop the `chisel_` prefix in the filenames? They are already contained within a `chisel` subdirectory after all.

File: userspace/chisel/chisel_table.h
Comment (jasondellaluce) [reply userspace/chisel/chisel_table.h]: Thanks for catching this! Just pushed another commit and put you in co-author! 🙏🏼

File: userspace/chisel/chisel_table.cpp
Comment (jasondellaluce) [reply userspace/chisel/chisel_table.cpp:172]: That's a great point. My goal for this PR was to introduce a no-op and just to relocate code parts with all the necessary renaming. What do you think about applying this changes in another PR after this one gets merged?

**Comment by jasondellaluce** [comment]
Thanks for your feedback @Molter73, that was super useful! 

As for the `cheisel_` prefix on files, I decided to put it in order to avoid ambiguity in project using chisels that right now are importing those files from libsinsp. This is kind of related to the broader discussion regarding header dependency ambiguities, which in this case is caused by the fact that both the `chisel` and `libsinsp` can potentially be used as includes. Once this gets merged, those imported `.h` files will not be available, which makes it much more explicit "which" parts of the code have been moved.

File: userspace/chisel/chisel_table.cpp
Comment (Molter73) [reply userspace/chisel/chisel_table.cpp:172]: I agree, tackling this in a separate PR is the right call.

File: userspace/chisel/chisel_table.cpp
Comment (Molter73) [reply userspace/chisel/chisel_table.cpp:20]: This makes sense, though I don't fully see why someone would use the `chisels` without `libsinsp`, which in turn makes me think of a discussion we had a couple of weeks ago in the community call about defining the public interface for the libs as a whole. For the time being, we can just leave the angled brackets.

**Review by Molter73 — DISMISSED** [review]
Don't know how much it's worth, but the PR approved in my opinion! Great job @jasondellaluce

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
I think this is safe to going in. Can you rebase @jasondellaluce ?

**Comment by leogr** [comment]
Closing and reopening to trigger the CI
/close

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**
